pkg/library/flatten/merge.go

```diff
@@ -118,3 +120,118 @@ func mergeVolume(into, from *dw.VolumeComponent) error {
 	}
 	return nil
 }
+
+func needsContainerContributionMerge(flattenedSpec *dw.DevWorkspaceTemplateSpec) bool {
+	hasContribution, hasTarget := false, false
+	for _, component := range flattenedSpec.Components {
+		if component.Container == nil {
+			// Ignore attribute on non-container components as it's not clear what this would mean
+			continue
+		}
+		if component.Attributes.GetBoolean(constants.ContainerContributionAttribute, nil) == true {
+			hasContribution = true
+		}
+		if component.Attributes.GetBoolean(constants.MergeContributionAttribute, nil) == true {
+			hasTarget = true
+		}
+	}
+	return hasContribution && hasTarget
+}
+
+func mergeContainerContributions(flattenedSpec *dw.DevWorkspaceTemplateSpec) error {
+	var contributions []dw.Component
+	for _, component := range flattenedSpec.Components {
+		if component.Container != nil && component.Attributes.GetBoolean(constants.ContainerContributionAttribute, nil) == true {
+			contributions = append(contributions, component)
+		}
+	}
+
+	var newComponents []dw.Component
+	mergeDone := false
+	for _, component := range flattenedSpec.Components {
+		if component.Container == nil {
+			newComponents = append(newComponents, component)
+			continue
+		}
+		if component.Attributes.GetBoolean(constants.ContainerContributionAttribute, nil) == true {
```

Comment:
Though this line (and other similar ones) could be just `if component.Attributes.GetBoolean(constants.ContainerContributionAttribute, nil) {...` I think it's okay to be explicit here about wanting the value of the `ContainerContributionAttribute` to be true.

Reply:
No, this is just me being a dummy :D 

I'll remove it.

Reply:
Fixed now, thanks :)